### PR TITLE
Adding DHW-status and showing the various states via the preset_mode parameter

### PR DESCRIPTION
**Pull request by bouwew**
With these changes the HA climate card shows this:
image

"Idle -  Weekschema": shows the heating state (idle,heating) and the active temperature setting.
When the schema-icon is green - mode Auto: show the selected schedule (e.g. Weekschema), or the preset-name when the selected schedule is temporary overridden by a preset, or "Temporary" when overridden by a manual temperature change.
When the heating-icon is red -  mode Heating: show only the preset-name when a preset is active.

Unfortunately, the HA climate-card does not support showing DHW (domestic_hot_water). A change request / update is needed for this. Will be requested.

The code is working, tested in my system. These updates need to use "my" haanna 0.11.3 version, see my pull-request for haanna. The code can probably use some cleanup/optimization, my programming-skills are a bit limited, sorry.

### CODE REVIEW

**Comment by bouwew** [comment]
Also, please note, when implementing showing the various states via the preset-name, I found there was a bug present in the Anna-xml-data. When a preset temperature is present in a schedule, the preset_state changes from none to the preset-name, but the preset_state is not cleared when the schedule changes back to a non-preset temperature setting. I had to work around this using indirect detection of the various states. The effect of this is that sometimes showing the correct change in state the climate card takes more than one scan_interval.

**Comment by CoMPaTech** [comment]
I will test & review, merge has to wait until https://github.com/laetificat/haanna/pull/11 and https://github.com/laetificat/haanna/pull/12 have merged and PyPi updated.

**Comment by CoMPaTech** [comment]
Just for future reference, we'll have to revisit the states with https://github.com/laetificat/haanna/pull/11 and cooling :)

**Comment by bouwew** [comment]
I assume @marcelveldt has made his own changes to the anna-ha code?
Maybe he can contribute his changes?

**Comment by marcelveldt** [comment]
> I assume @marcelveldt has made his own changes to the anna-ha code?
> Maybe he can contribute his changes?

https://github.com/marcelveldt/home-assistant/commit/37c33e4f05a1520a9e4a8b30e542722c1c83e070

**Comment by bouwew** [comment]
@marcelveldt, clear, thanks! Your system only supports cooling? Or also drying?
@all, I wonder, should we also add the HVAC_MODE_COOL or HVAC_MODE_HEAT_COOL constant as a possible attribute?

**Comment by CoMPaTech** [comment]
I'll have to try that, I recall from reading `components/climate/__init__.py` that it should become HVAC_MODE_HEAT_COOL, but not sure if that throws up to much when people don't have COOL

**Comment by marcelveldt** [comment]
> @marcelveldt, clear, thanks! Your system only supports cooling? Or also drying?
> @ALL, I wonder, should we also add the HVAC_MODE_COOL or HVAC_MODE_HEAT_COOL constant as a possible attribute?

It supports heating and cooling, only one of those modes can be active, selected by the thermostat based on the conditions.

From what I've seen in the code is that the state is now used as ON/OFF only based on the schedule state, not the heater state. ON= schedule running, OFF=manual control. So the state of the climate object in hass is not reflecting the boiler mode/state, that is just an additional property.

So in other words, you shouldn't add the HVAC_MODE_COOL as supported feature because it's currently not implemented like this, which is fine because you can't control the Anna's operating mode. Only the schedule.

**Review by bouwew — COMMENTED** [review]
This change in manifest.json should be ignored: 
image

**Comment by bouwew** [comment]
@marcelveldt, I noticed the same and made the change that you had in mind, please see my updates in the pull-request, of see my github: https://github.com/bouwew/anna-ha
My comment above was made with these changes already being implemented in mind.

**Comment by bouwew** [comment]
Actually, I was thinking, we could check for available capabilities first and make the code so that it automatically picks the applicable attributes.
Meaning, check for the presence of cooling_state and heating_state and decide based on presence whether to use HVAC_MODE_HEAT or HVAC_MODE_COOL or HVAC_MODE_HEAT_COOL.

**Comment by bouwew** [comment]
I'm working on some updates, adding the cooling stuff, etc.
I will create an improved pull-request.

**Comment by bouwew** [comment]
@CoMPaTech @marcelveldt 
I have updated the haanna and anna-ha code so that it automatically sets everything up according to the capabilities that are present. Please see my github: https://github.com/bouwew/anna-ha and
https://github.com/bouwew/haanna
You can link to my version of haanna by modifying `manifest.json` like this:
`"requirements": ["https://github.com/bouwew/haanna/archive/master.zip#haanna==0.11.6"]`
Can you please help testing this code?

**Comment by bouwew** [comment]
@CoMPaTech @marcelveldt
Sorry, I had forgotten to put my latest anna-ha update into my github.
That has been done just now.
Also, a few small improvements to haanna, my version is now 0.11.8.
